File: backend/PruebaTecnicaChileautos.Core/Filters/LocationFIlter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PruebaTecnicaChileautos.Core.Filters
{
    public class LocationFilter
    {
        public string? Name { get; set; }
        public string? Type { get; set; }
        public string? Dimension { get; set; }

        public string ToQueryString()
        {
            var queryParams = new List<string>();

            if (!string.IsNullOrWhiteSpace(Name))
                queryParams.Add($"name={Uri.EscapeDataString(Name)}");

            if (!string.IsNullOrWhiteSpace(Type))
                queryParams.Add($"name={Uri.EscapeDataString(Type)}");

            if (!string.IsNullOrWhiteSpace(Dimension))
                queryParams.Add($"name={Uri.EscapeDataString(Dimension)}");

            return string.Join("&", queryParams);
        }
    }
}

File: frontend/rick-and-morty-app/src/app/core/services/models/location-filter.ts
export class LocationFilter {
  name?: string;
  type?: string;
  dimension?: string = "";

  toQueryString(): string {
    const params: string[] = [];

    if (!!this.name) params.push(`name=${encodeURIComponent(this.name)}`);
    if (!!this.type) params.push(`type=${encodeURIComponent(this.type)}`);
    if (!!this.dimension) params.push(`dimension=${encodeURIComponent(this.dimension)}`);

    return params.join('&');
  }

   hasFilters() : boolean {
        return !!this.name || !!this.type || !!this.dimension;
    }
}